openedx/core/djangoapps/user_api/accounts/tests/test_views.py

```diff
@@ -0,0 +1,169 @@
+import unittest
+import ddt
+
+from django.test import TestCase
+from django.core.urlresolvers import reverse
+from django.conf import settings
+from rest_framework.test import APITestCase, APIClient
+
+from student.tests.factories import UserFactory
+from student.models import UserProfile
+
+TEST_PASSWORD = "test"
+
+@ddt.ddt
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class TestAccountAPI(APITestCase):
+
+    def setUp(self):
+        super(TestAccountAPI, self).setUp()
+
+        self.anonymous_client = APIClient()
+
+        self.bad_user = UserFactory.create(password=TEST_PASSWORD)
+        self.bad_client = APIClient()
+
+        self.staff_user = UserFactory(is_staff=True, password=TEST_PASSWORD)
+        self.staff_client = APIClient()
+
+        self.user = UserFactory.create(password=TEST_PASSWORD)
+        # Create some test profile values.
+        legacy_profile = UserProfile.objects.get(id=self.user.id)
+        legacy_profile.city = "Indi"
+        legacy_profile.country = "US"
+        legacy_profile.year_of_birth = 1900
+        legacy_profile.level_of_education = "m"
+        legacy_profile.goals = "world peace"
+        legacy_profile.save()
+        
+        self.accounts_base_uri = reverse("accounts_api", kwargs={'username': self.user.username})
```

Comment:
This confused me a little as it isn't the base_url but is the full URL with the user's name included in it.